### PR TITLE
#2418 Update KVM compatibility checks

### LINKED ISSUE #2418: Update the KVM capability checks.
Right now, the capabilities we check against are outdated -  https://github.com/firecracker-microvm/firecracker/blob/master/src/vmm/src/vstate/system.rs#L62

We need to iterate all KVM APIs currently used in Firecracker and update the capability list.

### PR DESCRIPTION
**Pull request by binoychitale**
Signed-off-by: binoychitale <binoychitale@gmail.com>

# Reason for This PR
Fixes #2418

## Description of Changes

- [ ] Update compatibility checks at startup to include missing KVM ioctls that are used in the vm

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

**Comment by binoychitale** [comment]
I believe the commit style checks are failing. I'll fix those

File: src/vmm/src/vstate/system.rs [hunk 1]
```diff
@@ -59,10 +59,27 @@ impl KvmContext {
...
+            VcpuEvents,
+            Xcrs,
+            Xsave,
+            ExtCpuid,
+        ];
 
         #[cfg(target_arch = "aarch64")]
-        let capabilities = vec![Irqchip, Ioeventfd, Irqfd, UserMemory, ArmPsci02];
+        let capabilities = vec![
+            Irqchip, Ioeventfd, Irqfd, UserMemory, ArmPsci02, DeviceCtrl, MpState,
```
Comment (dianpopa) [inline src/vmm/src/vstate/system.rs:81]: `OneReg` capability needs to be added for the KVM_GET_ONE_REG and KVM_SET_ONE_REG ioctl.

Comment (binoychitale) [reply src/vmm/src/vstate/system.rs:81]: I'll add that. 
Just for my info, where exactly are we using `KVM_GET_ONE_REG`  and `KVM_SET_ONE_REG`?
I see that we are using `KVM_GET_REGS` and `KVM_SET_REGS` via `VcpuFd::get_regs()` and `VcpuFd::set_regs()`
I wasnt able to find the specific API in the wrapper that maps to `KVM_{GET,SET}_ONE_REG`

Comment (dianpopa) [reply src/vmm/src/vstate/system.rs:81]: The get_one_reg functionality is the method used for retrieving the value of a register on aarch64. We are using it in multiple places: https://github.com/firecracker-microvm/firecracker/blob/17e259c2be64320a8d2d481813aefdd5f493b47c/src/arch/src/aarch64/regs.rs#L235, https://github.com/firecracker-microvm/firecracker/blob/17e259c2be64320a8d2d481813aefdd5f493b47c/src/arch/src/aarch64/regs.rs#L249. We use set_one_reg: https://github.com/firecracker-microvm/firecracker/blob/17e259c2be64320a8d2d481813aefdd5f493b47c/src/arch/src/aarch64/regs.rs#L385, https://github.com/firecracker-microvm/firecracker/blob/17e259c2be64320a8d2d481813aefdd5f493b47c/src/arch/src/aarch64/regs.rs#L164.

Comment (binoychitale) [reply src/vmm/src/vstate/system.rs:81]: Got it, thanks :)
I'm not sure if I was looking in the wrong place, or that these methods are not [documented](https://docs.rs/kvm-ioctls/0.6.0/kvm_ioctls/struct.VcpuFd.html)

I'll push the changes

**Comment by binoychitale** [comment]
CI pipeline failing with 
```
toomanyrequests: Rate exceeded
[Firecracker devtool] Error pulling docker image. Aborting.
```

File: src/vmm/src/vstate/system.rs (hunk 1 above)
Comment (binoychitale) [reply src/vmm/src/vstate/system.rs]: I think the documentation of non- `x86` components is not being rendered on [docs.rs](https://docs.rs/kvm-ioctls/0.6.0/i686-unknown-linux-gnu/kvm_ioctls/struct.VcpuFd.html)

The [build](https://docs.rs/about/builds) documentation on docs.rs states that the default compilation target is  `x86_64-unknown-linux-gnu`, which probably leads to the missing documentation for `aarch64` specific components.

I'll create an issue in the [upstream](https://github.com/rust-vmm/kvm-ioctls) to track this.

**Review by dianpopa — APPROVED** [review]

**Review by sandreim — APPROVED** [review]

**Pull request merged**